### PR TITLE
MM-23402 Fix for Mobile Web: Attach button only works with double tap on iOS

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
This was a regression was caused when we added conditional checks for mobile for preventing opening of file picket twice when clicked on attach button https://github.com/mattermost/mattermost-webapp/pull/4499.

The `isMobileApp` util func used here is not all that appropriate as it skips considering mobile browsers, i think it should have been `isMobile` util func(if we need a similar check In future we probably should check if touch listeners are available instead for reliability). 
Fixing this did not help as i think we need `onClick` to get around some of the security concerns in browsers with user initiated actions and trying to open file pickers programmatically. 

I resorted to passing `simulateClick` event to both `onClick` and `onTouchEnd` events and stop propagation as well as defaults to prevent them from opening the file picker twice. If touch events exist then it is fired first and the checks prevent click method from executing solving the root issue too.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23402

### CODE REVIEW

File: components/file_upload/file_upload.jsx [hunk 1]
```diff
@@ -565,15 +565,16 @@ class FileUpload extends PureComponent {
         this.setState({menuOpen: false});
     }
 
-    simulateInputClick = () => {
+    simulateInputClick = (e) => {
+        e.preventDefault();
+        e.stopPropagation();
```
Comment (ali-farooq0) [inline components/file_upload/file_upload.jsx]: i may be wrong but i remember looking into this at one point and attempting something similar but it still opened up the file picker twice for me. i.e. was seeing https://mattermost.atlassian.net/browse/MM-20939 behaviour. 

@sudheerDev can you please confirm if you're not able to reproduce the issue mentioned in MM-20939 with this change?

Comment (ali-farooq0) [reply components/file_upload/file_upload.jsx]: using chrome/ff

Comment (sudheerDev) [reply components/file_upload/file_upload.jsx]: yes, i am positive it does not cause a regression with the ticket you mentioned.

Comment (sudheerDev) [reply components/file_upload/file_upload.jsx]: Ahh, it does not cause that issue but it has not fixed the issue of picker not opening without double click. It worked lcaoclly for me. Gotto check this again

Comment (ali-farooq0) [reply components/file_upload/file_upload.jsx]: 👍

Comment (sudheerDev) [reply components/file_upload/file_upload.jsx]: @ali-farooq0 The PR works. I don't know what i was testing earlier. It fixes the ticket i linked and it does not cause a regression with the ticket you linked.

**Comment by sudheerDev** [comment]
waiting to see if this is needed for 5.22v

**Pull request merged**